modules/function/src/main/java/com/opengamma/strata/function/calculation/DefaultRatesMarketDataLookup.java

```diff
@@ -0,0 +1,263 @@
+import com.google.common.collect.Sets;
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.index.Index;
+import com.opengamma.strata.basics.market.MarketData;
+import com.opengamma.strata.basics.market.MarketDataKey;
+import com.opengamma.strata.basics.market.ObservableKey;
+import com.opengamma.strata.calc.CalculationRules;
+import com.opengamma.strata.calc.marketdata.CalculationEnvironment;
+import com.opengamma.strata.calc.marketdata.CalculationMarketData;
+import com.opengamma.strata.calc.marketdata.FunctionRequirements;
+import com.opengamma.strata.calc.runner.CalculationParameter;
+import com.opengamma.strata.collect.Messages;
+import com.opengamma.strata.market.id.SimpleCurveId;
+import com.opengamma.strata.market.key.IndexRateKey;
+import com.opengamma.strata.pricer.rate.RatesProvider;
+
+/**
+ * The rates lookup, used to select curves for pricing.
+ * <p>
+ * This provides access to discount curves and forward curves.
+ * <p>
+ * The lookup implements {@link CalculationParameter} and is used by passing it
+ * as an argument to {@link CalculationRules}. It provides the link between the
+ * data that the function needs and the data that is available in {@link CalculationMarketData}.
+ */
+@BeanDefinition(style = "light")
+final class DefaultRatesMarketDataLookup
+    implements RatesMarketDataLookup, ImmutableBean, Serializable {
+
+  /**
+   * The discount curves in the group, keyed by currency.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final ImmutableMap<Currency, SimpleCurveId> discountCurves;
+  /**
+   * The forward curves in the group, keyed by index.
+   */
+  @PropertyDefinition(validate = "notNull", builderType = "Map<? extends Index, SimpleCurveId>")
+  private final ImmutableMap<Index, SimpleCurveId> forwardCurves;
+
+  //-------------------------------------------------------------------------
+  /**
+   * Obtains an instance based on a map of discount and forward curve identifiers.
+   * <p>
+   * The discount and forward curves refer to the curve identifier.
+   * The curves themselves are provided in {@link CalculationEnvironment}
+   * using {@link SimpleCurveId} as the identifier.
+   * 
+   * @param discountCurveIds  the discount curve identifiers, keyed by currency
+   * @param forwardCurveIds  the forward curves identifiers, keyed by index
+   * @return the rates lookup containing the specified curves
+   */
+  public static DefaultRatesMarketDataLookup of(
+      Map<Currency, SimpleCurveId> discountCurveIds,
+      Map<? extends Index, SimpleCurveId> forwardCurveIds) {
+
+    return new DefaultRatesMarketDataLookup(discountCurveIds, forwardCurveIds);
+  }
+
+  //-------------------------------------------------------------------------
+  @Override
+  public ImmutableSet<Currency> getDiscountCurveCurrencies() {
+    return discountCurves.keySet();
+  }
+
+  @Override
+  public ImmutableSet<MarketDataKey<?>> getDiscountCurveIds(Currency currency) {
```

Comment:
In theory, the lookup could be based on `DiscountFactors` rather than a curve, or on data extracted from a surface. ie. the interface is deliberately designed to not commit the implementation to being a curve. (same as is done for swaptions, where volatilities might be from a surface, cube or something else). Probably the method name is wrong.